plugins/org.komodo.relational/src/org/komodo/relational/internal/RelationalObjectImpl.java

```diff
@@ -34,6 +35,19 @@ protected RelationalObjectImpl( final Repository repository,
     /**
      * {@inheritDoc}
      *
+     * @see org.komodo.repository.ObjectImpl#setPrimaryType(org.komodo.spi.repository.Repository.UnitOfWork, java.lang.String)
+     * @throws UnsupportedOperationException
+     *         if called
+     */
+    @Override
+    public final void setPrimaryType( final UnitOfWork uow,
+                                      final String typeName ) throws UnsupportedOperationException {
+        throw new UnsupportedOperationException();
```

Comment:
Fair enough. Do you think you could either replace with a KException and keep the throw clause or keep the UOE and remove the throw clause?